src/app/new-char/page.tsx

```diff
@@ -2,18 +2,23 @@
 
 import { Form, Wrapper } from '@/app/style';
 import { Button } from '@/app/player/[characterId]/[state]/style';
-import React from 'react';
+import React, { useState } from 'react';
 import useRealtimeState from '@/app/hooks/useRealtimeState';
 import type { pjData } from '@/@types/pjData';
 
 const characterCreation = () => {
   const [players, setPlayers] = useRealtimeState<pjData[]>();
+  const [buttonDisabled, setButtonDisabled] = useState<boolean>(false);
 
-  const sendToServer = (formData: FormData) => {
+  const handleCharacterCreation = () => {
+    setButtonDisabled(true);
+  };
```

Comment:
Passa a chamada do `sendToServer` pra cá, assim você mantem o action como apenas o handler, e mantém a única responsabilidade do `sendToServer` de enviar pro servidor, sem ter vínculo com o bloqueio do botão.
```suggestion
  const handleCharacterCreation = (formData: FormData) => {
    setButtonDisabled(true);
    sendToServer(formData);
    setButtonDisabled(false);
  };
```